src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -8198,6 +8250,401 @@ public void ParameterScope_DefaultOnParameter(string keyword, bool shadow, bool
             AssertParameterScope(keyword, shadow, isRecord, flags, source);
         }
 
+        [Fact]
+        public void ParameterScope_TypeParameterShadows_01()
+        {
+            var src = @"
+class C1<T>(T a, int T) : Base(T)
+{
+    T A = a;
+    int T1 = T;
+    event System.Action T2 = T;
+    int T3 {get;} = T;
+
+    void M()
+    {
+        var x = T;
+        T++;
+    }
+}
+
+class Base
+{
+    public Base(int x){}
+}
+";
+            var comp = CreateCompilation(src);
+
+            // Even though specification for the feature says that enclosing type type parameter should be found before 
+            // primary constructor parameter, even in field initializers and base clause arguments, we already
+            // have pre-existing behavior with records that is not consistent with that. Specifically, primary constructor
+            // parameters come first in field initializers and base clause arguments for records.
+            // Keeping this behavior for now.
```

Comment:
Perhaps worth mentioning this in the speclet?

Reply:
> Perhaps worth mentioning this in the speclet?

I am going to bring this as an open design question to LDM.